cou/steps/plan.py

```diff
@@ -503,7 +507,11 @@ def _get_purge_data_steps(analysis_result: Analysis, args: CLIargs) -> list[PreU
         return [
             PreUpgradeStep(
                 description=msg,
-                coro=purge(analysis_result.model, before=args.purge_before),
+                coro=purge(
+                    analysis_result.model,
+                    analysis_result.apps_data_plane,
+                    before=args.purge_before,
+                ),
```

Comment:
**nitpick (blocking):** 

Please pass the required application only:

```
coro=purge(
    analysis_result.model,
    get_applications_by_charm_name(analysis_result.apps_data_plane, "nova-cloud-controller")
    before=args.purge_before,
),
```

Reply:
now it makes sense, with your following comment

Reply:
Can we make this out of this PR? This changes the logic of existing code, and breaks many unit test and mock test

Reply:
Ok, please leave a comment on the code and we can resolve this one.